app/scripts/lib/createConfirmationValidationMiddleware.ts

```diff
@@ -0,0 +1,53 @@
+import { errorCodes } from 'eth-rpc-errors';
+import { isValidAddress } from 'ethereumjs-util';
+import { isValidHexAddress, JsonRpcRequest } from '@metamask/utils';
+import { JsonRpcMiddleware } from 'json-rpc-engine';
+
+import { MESSAGE_TYPE } from '../../../shared/constants/app';
+import { parseTypedDataMessage } from '../../../shared/modules/transaction.utils';
+
+import { EIP712_PRIMARY_TYPE_PERMIT } from '../../../shared/constants/transaction';
+
+/**
+ * Returns a middleware that validated incoming confirmations
+ */
+
+export default function createConfirmationValidationMiddleware(): JsonRpcMiddleware<
```

Comment:
Rather than adding validation specific to the extension, would it be more effective to add this to the existing validations in the wallet middleware defined in `@metamask/eth-json-rpc-middleware`?

https://github.com/MetaMask/eth-json-rpc-middleware/blob/main/src/wallet.ts#L292

This would help to centralise the validations in one place, and align with the long-term vision to move all validation to the middleware via a separate package, so it's no longer a client specific concern.